src/routes/NoticesList/index.tsx

```diff
@@ -19,33 +20,86 @@ export default function NoticesListPage() {
   const [noticeQueryParams, setNoticeQueryParams] = React.useState<{
     paidByMe?: boolean;
     registeredToMe?: boolean;
-  }>({});
+    /** continuation token, used to paginate the elements */
+    continuationToken: string;
+    /** date order, the only one avaiable. default DESC */
+    ordering: 'ASC' | 'DESC';
+  }>({ continuationToken: '', ordering: 'DESC' });
+
   const [activeTab, setActiveTab] = React.useState(NoticesTabs.all);
+  const [currentPage, setCurrentPages] = React.useState(0);
+  const [pages, setPages] = React.useState(['']);
 
   const { t } = useTranslation();
+
   const noticesList = useNormalizedNoticesList(noticeQueryParams);
 
   const {
     queryResult: { data, error, refetch }
   } = noticesList;
 
-  const onChange = (activeTab: NoticesTabs) => {
+  const resetPagination = () => {
+    setCurrentPages(0);
+    setPages(['']);
+  };
+
+  const toggleDateOrder = () => {
+    resetPagination();
+    const newDateOrdering = noticeQueryParams.ordering === 'DESC' ? 'ASC' : 'DESC';
+    setNoticeQueryParams({
+      ...noticeQueryParams,
+      ordering: newDateOrdering,
+      continuationToken: ''
+    });
+  };
+
+  const onTabChange = (activeTab: NoticesTabs) => {
+    resetPagination();
     setActiveTab(activeTab);
     switch (activeTab) {
       case NoticesTabs.all:
-        setNoticeQueryParams({});
+        setNoticeQueryParams({
+          ordering: 'DESC',
+          continuationToken: '',
+          registeredToMe: undefined,
+          paidByMe: undefined
+        });
         break;
       case NoticesTabs.paidByMe:
-        setNoticeQueryParams({ paidByMe: true });
+        setNoticeQueryParams({
+          ordering: 'DESC',
+          continuationToken: '',
+          paidByMe: true,
+          registeredToMe: undefined
+        });
         break;
       case NoticesTabs.registeredToMe:
-        setNoticeQueryParams({ registeredToMe: true });
+        setNoticeQueryParams({
+          ordering: 'DESC',
+          continuationToken: '',
+          registeredToMe: true,
+          paidByMe: undefined
+        });
     }
   };
 
+  const goToPage = (direction: number) => {
+    const pageIndex = currentPage + direction;
+    setCurrentPages(pageIndex);
+    setNoticeQueryParams({ ...noticeQueryParams, continuationToken: pages[pageIndex] });
+  };
+
   React.useEffect(() => {
-    refetch();
-  }, [activeTab]);
+    (async () => {
+      const response = await refetch();
+      const continuationToken = response.data?.continuationToken;
+      // if no token is returned we reach the end
+      if (!continuationToken) return;
+      // is a new token? if not, no need to update the token pages array
+      const isNewToken = !pages.find((oldToken) => oldToken === continuationToken);
+      if (isNewToken) setPages([...pages, continuationToken]);
```

Comment:
I would change this line to the following, so to avoid any race conditions and be sure that the latest pages state is used.
```js
if (isNewToken) setPages((prevPages) => [...prevPages, continuationToken]);
```